cocotb/share/lib/fli/FliObjHdl.cpp

```diff
@@ -357,30 +406,93 @@ int FliLogicObjHdl::set_signal_value(const int32_t value,
         }
 
         if (m_is_var) {
-            mti_SetVarValue(get_handle<mtiVariableIdT>(), (mtiLongT)m_mti_buff);
+            switch (action) {
+                case GPI_DEPOSIT:
+                    mti_SetVarValue(get_handle<mtiVariableIdT>(),
+                                    (mtiLongT)m_mti_buff);
+                    return 0;
+                case GPI_FORCE:
+                    LOG_ERROR("Cannot force a VHDL variable");
+                    return -1;
+                case GPI_RELEASE:
+                    LOG_ERROR("Cannot release a VHDL variable");
+                    return -1;
+                default:
+                    LOG_ERROR("Unknown set value action (%d)", action);
+                    return -1;
+            }
         } else {
-            mti_SetSignalValue(get_handle<mtiSignalIdT>(),
-                               (mtiLongT)m_mti_buff);
+            switch (action) {
+                case GPI_DEPOSIT:
+                    mti_SetSignalValue(get_handle<mtiSignalIdT>(),
+                                       (mtiLongT)m_mti_buff);
+                    return 0;
+                case GPI_FORCE: {
+                    std::string value_str = "2#";
```

Comment:
You could also use `10#<val>` and `-10#<val>`. Questa won't make a difference, since it will create an int from it anyway. Not sure which is better, just a note :).

Reply:
Does the force command accepts negative numbers for setting std_logic_vectors?